crates/aptos/src/common/init.rs

```diff
@@ -189,7 +179,105 @@ impl CliCommand<()> for InitTool {
+            rest_url.clone()
+        } else {
+            eprintln!(
+                "Enter your rest endpoint [Current: {} | No input: {}]",
+                profile_config.rest_url.as_deref().unwrap_or("None"),
+                DEFAULT_REST_URL
+            );
+            let input = read_line("Rest endpoint")?;
+            let input = input.trim();
+            if input.is_empty() {
+                eprintln!("No rest url given, using {}...", DEFAULT_REST_URL);
+                reqwest::Url::parse(DEFAULT_REST_URL).map_err(|err| {
+                    CliError::UnexpectedError(format!("Failed to parse default rest URL {}", err))
+                })?
+            } else {
+                reqwest::Url::parse(input)
+                    .map_err(|err| CliError::UnableToParse("Rest Endpoint", err.to_string()))?
+            }
+        };
+        profile_config.rest_url = Some(rest_url.to_string());
+
+        // Faucet Endpoint
+        let faucet_url = if self.skip_faucet {
+            eprintln!("Not configuring a faucet because --skip-faucet was provided");
+            None
+        } else if let Some(ref faucet_url) = self.faucet_url {
+            eprintln!("Using command line argument for faucet URL {}", faucet_url);
+            Some(faucet_url.clone())
+        } else {
+            eprintln!(
+                "Enter your faucet endpoint [Current: {} | No input: {} | 'skip' to not use a faucet]",
+                profile_config
+                    .faucet_url.as_deref()
+                    .unwrap_or("None"),
+                DEFAULT_FAUCET_URL
+            );
+            let input = read_line("Faucet endpoint")?;
+            let input = input.trim();
+            if input.is_empty() {
+                eprintln!("No faucet url given, using {}...", DEFAULT_FAUCET_URL);
+                Some(reqwest::Url::parse(DEFAULT_FAUCET_URL).map_err(|err| {
+                    CliError::UnexpectedError(format!("Failed to parse default faucet URL {}", err))
+                })?)
+            } else if input.to_lowercase() == "skip" {
+                eprintln!("Skipping faucet");
+                None
+            } else {
+                Some(
+                    reqwest::Url::parse(input).map_err(|err| {
+                        CliError::UnableToParse("Faucet Endpoint", err.to_string())
+                    })?,
+                )
+            }
+        };
+        profile_config.faucet_url = faucet_url.as_ref().map(|inner| inner.to_string());
+        Ok(())
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum Network {
+    Testnet,
+    Devnet,
+    Local,
+    Custom,
+}
+
+impl FromStr for Network {
+    type Err = CliError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s.to_lowercase().trim() {
+            "testnet" => Self::Testnet,
+            "devnet" => Self::Devnet,
+            "local" => Self::Local,
+            "custom" => Self::Custom,
```

Comment:
can we not leverage the chain_id type? I feel those should probably be adjusted too....

Reply:
Yes, we can leverage ChainId, except that devnet changes every week which makes it hard to use here